daffodil-core/src/main/scala/org/apache/daffodil/core/grammar/primitives/PrimitivesExpressions.scala

```diff
@@ -197,6 +202,7 @@ case class SetVariable(stmt: DFDLSetVariable, override val term: Term)
 
   override lazy val exprText = stmt.value
   override lazy val exprNamespaces = stmt.xml.scope
+  override lazy val exprTargetNamespace = stmt.annotatedSC.targetNamespace
```

Comment:
This is pretty paindful that everywhere we have xyzNamespaces to get the scopes we have in parallel xyzTargetNamespace also. 

These should be one context object, not two both of which must be maintained in parallel in all these different places. 

I.e., one tuple of these two things, not two separate things passed all the time.

Reply:
Agreed, especially if the solution for DAFFODIL-2917 is to add support for xpathDefaultNamespace, which would be another piece of context for resolving things. And maybe the value of unqualifiedPathStepPolicy (which may or may not go away?) also wants to be part of that. Should we add the `XSDContext` thing you suggest and put all those pieces in it? And anything that resolves QNames needs an XSDContext, and any new context needed for such resolution is added to that object?

Reply:
I suggest merging this as is. There are a variety of other namespace-related things I'd like to consider before designing the complete solution.  Minimization of namespace bindings is one of them, for example, which involves assembling more than one such context and potentially creating unique namespace prefix bindings to resolve ambiguity.